index.js

```diff
@@ -1095,10 +1098,15 @@ Multi.prototype.exec = function (callback) {
                     args[args.length - 1](null, reply);
                 }
             }
+        } else if (err) {
+            err.code = 'EXECABORT';
+            err.errors = errors;
         }
 
         if (callback) {
-            callback(null, replies);
+            callback(err, replies);
```

Comment:
I read the [transactions] (http://redis.io/topics/transactions) part again and it seems like this is a special behavior by redis.
The transaction is only aborted if any command can't be queued. If the rough syntax is fine and redis queues the commands without any error all commands are going to be executed. If any of these commands fail it'll return an error individually. But this is not going to abort the transaction. So all commands that got executed are indeed executed in the order of the occurrence.

Earlier you got an array of errors if a command could not be queued. This is changed to be a single error with the `EXECABORT` code and the failing command as element of `err.errors`.
If a single command could not be executed this would not have returned an error in the error return value but in the reply. 
That way it's not obvious that commands could not be executed. And since commands will only fail because of programmer errors this should result in an error instead. So I changed it to return both: the err, to indicate that something went wrong with all not executed commands inserted in `err.errors` including the execution position and the reply with all returned execution values that passed and undefined values for errors.
It might be an idea to keep the errors in the reply as well instead of passing undefined here. What do you think @bcoe @brycebaril @bitinn?